### PR TITLE
Add ContentTypeForm functionality to DecodeForm

### PR DESCRIPTION
**Pull request by syntaqx**
Some of my companies legacy applications require the ability to use `ContentTypeForm`. I didn't realize go-chi/render didn't support this until we started running into random breakage..

### CODE REVIEW

**Comment by syntaqx** [comment]
Literally I just want the ability for decoding the following:

```
$ curl -d "name=test" -X POST http://mysite.com/create
```

I would say we could spend the time to recreate the functionality of gorilla/schema if you're trying to keep this repository free of dependencies, but it's pretty much the defacto standard for dealing with form data, and has fixed a lot of edge cases around them.

**Comment by VojtechVitek** [comment]
1. https://github.com/google/go-querystring
2. https://github.com/gorilla/schema
3. https://github.com/ajg/form

We have couple options. Seems like 1. is the most mature.

**Comment by syntaqx** [comment]
If I'm not mistaken, [go-querystring](https://github.com/google/go-querystring) is specifically for the case of utilizing urlencoded query arguments, rather than extracting form data into url.Values.

When comparing [ajg/form](https://github.com/ajg/form) and [gorilla/schema](https://github.com/gorilla/schema), it seems like both implementations could be viable, it's honestly a flip of the coin on that one.

**Comment by syntaqx** [comment]
@VojtechVitek How would you like me to proceed with this? I believe that gorilla/schema is a good choice for this, but if you would like to vet the others, that's reasonable.

**Comment by VojtechVitek** [comment]
@syntaqx you're right, go-querystring only implements encoding.

We should evaluate decide between the two other packages. Both seem very mature to me.

However, https://github.com/ajg/form looks simpler and more elegant to me.

gorilla/schema looks quite complex for what it's supposed to do, especially the internal cache: https://github.com/gorilla/schema/blob/master/cache.go#L18-L33

**Comment by syntaqx** [comment]
Sorry for the massive delay on this. I've switched to using `ajg/form`. Please let me know if you have additional concerns.

**Comment by syntaqx** [comment]
@VojtechVitek Let me know if you need anything from me. Would love to switch over to using this package instead of a fork again!

**Comment by VojtechVitek** [comment]
@syntaqx thanks, I'll take a look when I have some spare time. I'm too busy right now with my job

File: decoder.go [hunk 1]
```diff
@@ -7,6 +7,8 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+
+	"github.com/ajg/form"
```
Comment (VojtechVitek) [inline decoder.go]: should we bring in github.com/ajg/form as golang/dep dependency?

Comment (syntaqx) [reply decoder.go]: I think that's a more global question for you, for the project. Currently there's no dependency tooling, so I didn't add any.

**Comment by otraore** [comment]
Any reason as to why this isn't merged yet, we need this at work too.

**Comment by VojtechVitek** [comment]
@otraore I want to get this merged together with golang/dep files.

**Comment by syntaqx** [comment]
I've added dep.

**Comment by syntaqx** [comment]
Any follow up on this?

**Comment by VojtechVitek** [comment]
Sorry I'm too busy at this point. I'll try to come back to this when I have some spare time.

**Comment by syntaqx** [comment]
Hey @VojtechVitek - I know you've been busy, but was just wondering if you've had the chance to look at this? If not, perhaps you'd be willing to add myself or someone else as a maintainer? Would love to be able to keep using this, but without support it's becoming very unreasonable.

**Comment by syntaqx** [comment]
As an aside, `gorilla/schema` seems to be a better package overall, but I think the general purpose of this PR can use either. Seems that there are occasional reflection issues with `ajg/form` where it has issues deciphering things, but that can be overridden on a per-application basis, rather than needing to be packaged as the primary option.

**Comment by SuperFluffy** [comment]
Is there some progress on this? I'd love to see this land.

**Comment by syntaqx** [comment]
¯\_(ツ)_/¯ can't seem to get him to merge it.

**Comment by VojtechVitek** [comment]
Open-source is tough, man. Merging this PR means I'd have to spend hours of testing the changeset in the pre-production and production environments with our projects.

I'm willing to do that, but I can't do that right away, since I have different priorities at this time.

Send bitcoin or ETH and I'll find some time sooner! :D

**Comment by VojtechVitek** [comment]
You're right. I'm slow. And I'm sorry.

I'll try to test this when I find some time soonish.

**Comment by syntaqx** [comment]
Happy 2019 @VojtechVitek - How're we doing on getting this merged friend? Especially given we have Semantic Versioning now, I feel like the issue is less complicated.

**Comment by pzeinlinger** [comment]
Hey @VojtechVitek,
what about merging this PR? It's not complicated, and we have been running this for quite some time now in production. No issues at all.

**Comment by FryDay** [comment]
Is this project no longer maintained? This is a pretty simple PR and has been open for 4 years.

**Comment by VojtechVitek** [comment]
I'm sorry. I don't currently use go-chi/render.

This PR has some old module mgmt system, so I prefer not to merge it right away.

**Comment by syntaqx** [comment]
I'm more than happy to include Go modules, assuming that you will actually make an effort to get this merged.

What are you using instead of this project, so that people can look to that?

Additionally, I'm more than happy to help maintain this project if you're open to it. It certainly needs a few modernization updates, but there's plenty of participation on the project that could use responses, and some additional features would make it a more than viable option for folks.and would be more than happy to even maintain the repository to provide it with some additional functionality that it would benefit from, b

**Comment by syntaqx** [comment]
@VojtechVitek I've updated the pull request:

- Uses go modules
- Swapped Travis for GitHub Actions (the old tests relied on goimports, and given that I don't use Travis I just swapped to actions instead) - Let me know if you have any concerns here.
- Added a couple of everyone's favorite change: build badges in the readme

If we could move to getting this merged in, that would be awesome. Also, please let me know your thoughts on the future of this project, as I would be more than happy to provide it a bit of attention and some much needed updates, but I don't want another 4 years of PR delays ;)

**Comment by syntaqx** [comment]
Sorry, couple minor updates there; didn't realize the linting wasn't passing. Actually good for review/merge now.

**Comment by FryDay** [comment]
I would be willing to assist @syntaqx in maintaining Render. Chi and Render are certainly being used in production in various projects and really need to be kept current.

File: responder.go [hunk 2]
```diff
@@ -66,7 +66,7 @@ func PlainText(w http.ResponseWriter, r *http.Request, v string) {
 	if status, ok := r.Context().Value(StatusCtxKey).(int); ok {
 		w.WriteHeader(status)
 	}
-	w.Write([]byte(v))
+	w.Write([]byte(v)) //nolint:errcheck
```
Comment (VojtechVitek) [inline responder.go:69]: nit: `_ = w.Write()` would be imho more idiomatic than using some special comments

**Review by VojtechVitek — APPROVED** [review]
LGTM

Thanks for switching to Go Modules and to Github Actions too. Great job!

For reference:
https://github.com/syntaqx/go-chi-render/runs/2451985236
https://github.com/syntaqx/go-chi-render/runs/2451985149

**Pull request merged**